packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -645,11 +645,13 @@ export const optionalFieldEditor: OptionalFieldEditor = {
 		},
 	}),
 
-	buildChildChange: (index: number, childChange: NodeId): OptionalChangeset => {
-		assert(index === 0, 0x404 /* Optional fields only support a single child node */);
+	buildChildChanges: (changes: Iterable<[number, NodeId]>): OptionalChangeset => {
 		return {
```

Comment:
Before returning, it would make sense to assert that `childChanges.length <= 1`.